pinot-core/src/main/java/org/apache/pinot/core/query/request/context/utils/QueryContextUtils.java

```diff
@@ -41,6 +41,15 @@ public static boolean isSelectionQuery(QueryContext query) {
     return query.getAggregationFunctions() == null;
   }
 
+  /**
+   * Returns {@code true} if the given query is a simple selection query, {@code false} otherwise.
+   *
+   * Simple selection query at this moment means selection query without order-by.
+   */
+  public static boolean isSimpleSelectionQuery(QueryContext query) {
```

Comment:
(minor) we usually call it selection-only query
```suggestion
  public static boolean isSelectionOnlyQuery(QueryContext query) {
```